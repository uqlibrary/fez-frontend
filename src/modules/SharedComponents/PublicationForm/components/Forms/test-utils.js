import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
