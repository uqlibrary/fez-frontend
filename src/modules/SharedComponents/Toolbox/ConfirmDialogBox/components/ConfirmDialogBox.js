import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import DialogActions from '@mui/material/DialogActions';
import DialogContentText from '@mui/material/DialogContentText';
import DialogContent from '@mui/material/DialogContent';
import withStyles from '@mui/styles/withStyles';
import Grid from '@mui/material/Grid';

export const styles = theme => ({
    alternateActionButtonClass: {
        color: ((theme.palette || {}).white || {}).main,
        backgroundColor: ((theme.palette || {}).warning || {}).main,
        '&:hover': {
            backgroundColor: ((theme.palette || {}).warning || {}).dark,
        },
    },
});

export class ConfirmDialogBox extends Component {
    static propTypes = {
        className: PropTypes.string,
        classes: PropTypes.object,
        confirmDialogBoxId: PropTypes.string,
        hideCancelButton: PropTypes.bool,
        locale: PropTypes.object,
        onAction: PropTypes.func,
        onCancelAction: PropTypes.func,
        onAlternateAction: PropTypes.func,
        onRef: PropTypes.func,
        showAlternateActionButton: PropTypes.bool,
    };

    static defaultProps = {
        hideCancelButton: false,
        locale: {
            confirmationTitle: 'Confirmation',
            confirmationMessage: 'Are you sure?',
            cancelButtonLabel: 'No',
            confirmButtonLabel: 'Yes',
        },
        showAlternateActionButton: false,
        confirmDialogBoxId: 'dialog-box',
    };

    constructor(props) {
        super(props);

        this._onCancelAction = this._onCancelAction.bind(this);
        this._onAlternateAction = this._onAlternateAction.bind(this);
        this._hideConfirmation = this._hideConfirmation.bind(this);
        this._onAction = this._onAction.bind(this);

        this.state = {
            isDialogOpen: false,
        };
    }

    componentDidMount() {
        this.props.onRef(this);
    }

    componentWillUnmount() {
        this.props.onRef(undefined);
    }

    showConfirmation() {
        this.setState({
            isDialogOpen: true,
        });
    }

    _hideConfirmation() {
        this.setState({
            isDialogOpen: false,
        });
    }

    _onAction() {
        this._hideConfirmation();
        this.props.onAction && this.props.onAction();
    }

    _onCancelAction() {
        this._hideConfirmation();
        !!this.props.onCancelAction && this.props.onCancelAction();
    }

    _onAlternateAction() {
        this._hideConfirmation();
        !!this.props.onAlternateAction && this.props.onAlternateAction();
    }

    render() {
        const { classes } = this.props;
        return (
            <Dialog style={{ padding: 6 }} open={this.state.isDialogOpen}>
                <DialogTitle data-testid="message-title">{this.props.locale.confirmationTitle}</DialogTitle>
                <DialogContent>
                    <DialogContentText data-testid="message-content">
                        {this.props.locale.confirmationMessage}
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Grid container spacing={1}>
                        <Grid item xs sx={{ display: { xs: 'none', sm: 'block' } }} />

                        <Grid item xs={12} sm={'auto'}>
                            <Button
                                children={this.props.locale.confirmButtonLabel}
                                autoFocus
                                color={'primary'}
                                fullWidth
                                onClick={this._onAction}
                                data-analyticsid={`confirm-${this.props.confirmDialogBoxId}`}
                                data-testid={`confirm-${this.props.confirmDialogBoxId}`}
                            />
                        </Grid>
                        {this.props.showAlternateActionButton && (
                            // an optional middle button that will display in a warning colour
                            <Grid item xs={12} sm={'auto'}>
                                <Button
                                    variant={'contained'}
                                    data-analyticsid={`alternate-${this.props.confirmDialogBoxId}`}
                                    data-testid={`alternate-${this.props.confirmDialogBoxId}`}
                                    className={classes.alternateActionButtonClass}
                                    children={this.props.locale.alternateActionButtonLabel}
                                    fullWidth
                                    onClick={this._onAlternateAction}
                                />
                            </Grid>
                        )}
                        {!this.props.hideCancelButton && (
                            <Grid item xs={12} sm={'auto'}>
                                <Button
                                    variant={'contained'}
                                    color={'primary'}
                                    data-analyticsid={`cancel-${this.props.confirmDialogBoxId}`}
                                    data-testid={`cancel-${this.props.confirmDialogBoxId}`}
                                    children={this.props.locale.cancelButtonLabel}
                                    fullWidth
                                    onClick={this._onCancelAction}
                                />
                            </Grid>
                        )}
                    </Grid>
                </DialogActions>
            </Dialog>
        );
    }
}

export default withStyles(styles)(ConfirmDialogBox);
