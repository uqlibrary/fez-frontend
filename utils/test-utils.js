/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';

import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';

import mediaQuery from 'css-mediaquery';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@testing-library/dom';
import preview from 'jest-preview';

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <LocalizationProvider dateAdapter={AdapterMoment}>{props.children}</LocalizationProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => {
    return {
        user: userEvent.setup(),
        ...render(ui, { wrapper: AllTheProviders, ...options }),
    };
};

export const WithRouter = ({ children, route = '/', initialEntries = [route] }) => {
    const routes = [{ path: route, element: children }];
    const router = createMemoryRouter(routes, { initialEntries: initialEntries });
    return <RouterProvider router={router} />;
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const assertTooltipText = async (trigger, tooltipText) => {
    expect(trigger).toBeInTheDocument();
    await userEvent.hover(trigger);
    await waitFor(() => {
        expect(screen.getByRole('tooltip', { name: tooltipText, hidden: true })).toBeVisible();
    });
};

const extensionToMimeMap = {
    tiff: 'image/tiff',
    wav: 'audio/wave',
    zip: 'application/zip',
};
const mimeType = filename => extensionToMimeMap[filename.split('.').pop()];

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};

export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(datastream => {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mimeType(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(prefix => {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_id: index + 1,
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, datastreams) =>
    datastreams.find(datastream => datastream.dsi_dsid === filename);

export const createMatchMedia = width => {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        /* istanbul ignore next */
        addListener: () => {},
        /* istanbul ignore next */
        removeListener: () => {},
    });
};

export const getFilenameExtension = filename => filename.split('.').pop();
export const getFilenameBasename = filename =>
    filename.replace(new RegExp(`/\.${getFilenameExtension(filename)}$/`), '');
export const addFilesToFileUploader = files => {
    const { screen, fireEvent } = reactTestingLib;
    // create a list of Files
    const fileList = files.map(file => {
        // if file it's a string, treat it as a filename
        if (typeof file === 'string') {
            return new File([getFilenameBasename(file)], file, { type: `image/${getFilenameExtension(file)}` });
        }
        // otherwise expect it to be a object with filename and mimeType keys
        return new File([getFilenameBasename(file.filename)], file.filename, { type: file.mimeType });
    });
    // drag and drop files
    fireEvent.drop(screen.getByTestId('fez-datastream-info-input'), {
        dataTransfer: {
            files: fileList,
            types: ['Files'],
        },
    });
};
export const setFileUploaderFilesToClosedAccess = async (files, timeout = 500) => {
    const { fireEvent } = reactTestingLib;
    // set all files to closed access
    for (const file of files) {
        const index = files.indexOf(file);
        await waitFor(() => screen.getByText(new RegExp(getFilenameBasename(file))), { timeout });
        fireEvent.mouseDown(screen.getByTestId(`dsi-open-access-${index}-select`));
        fireEvent.click(screen.getByRole('option', { name: 'Closed Access' }));
    }
};

export const assertEnabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeEnabled());
export const assertDisabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeDisabled());

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    withRedux,
    AllTheProviders,
    WithReduxStore,
    assertTooltipText,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
    createMatchMedia,
    preview,
    userEvent,
    assertEnabled,
    assertDisabled,
    getFilenameExtension,
    getFilenameBasename,
    addFilesToFileUploader,
    setFileUploaderFilesToClosedAccess,
};
