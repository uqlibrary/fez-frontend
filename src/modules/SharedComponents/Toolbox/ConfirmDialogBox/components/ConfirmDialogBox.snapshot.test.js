import React from 'react';
import ConfirmDialogBox, { ConfirmDialogBox as ConfirmDialogBoxClass } from './ConfirmDialogBox';
import { rtlRender, fireEvent } from 'test-utils';

const defaultProps = {
    classes: {},
    hideCancelButton: false,
    locale: {
        alternateActionButtonLabel: 'Maybe',
        confirmationTitle: 'Confirmation',
        confirmationMessage: 'Are you sure?',
        cancelButtonLabel: 'No',
        confirmButtonLabel: 'Yes',
    },
    onAction: jest.fn(),
    onCancelAction: jest.fn(),
    onAlternateAction: jest.fn(),
    showAlternateActionButton: false,
    isOpen: true,
};

function setup(testProps = {}, renderer = rtlRender) {
    const props = { ...defaultProps, ...testProps };
    return renderer(<ConfirmDialogBox {...props} />);
}

describe('ConfirmDialogBox snapshots tests', () => {
    it('renders component with yes/no buttons', () => {
        const { container, getByTestId, getByRole } = setup();
        expect(getByTestId('message-title')).toBeInTheDocument();
        expect(getByTestId('message-content')).toBeInTheDocument();
        expect(getByRole('button', { name: /Yes/ })).toBeInTheDocument();
        expect(getByRole('button', { name: /No/ })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('renders component with yes', () => {
        const { container, getByRole, queryByRole } = setup({ hideCancelButton: true });
        expect(getByRole('button', { name: /Yes/ })).toBeInTheDocument();
        expect(queryByRole('button', { name: /No/ })).not.toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('renders component with yes/no/maybe buttons', () => {
        const { container, getByRole } = setup({ showAlternateActionButton: true });
        expect(getByRole('button', { name: /Yes/ })).toBeInTheDocument();
        expect(getByRole('button', { name: /Maybe/ })).toBeInTheDocument();
        expect(getByRole('button', { name: /No/ })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('renders component with customised locale', () => {
        const { container, getByRole, getByText } = setup({
            locale: {
                alternateActionButtonLabel: 'ENG: Maybe',
                cancelButtonLabel: 'ENG: No',
                confirmationTitle: 'ENG: Confirmation',
                confirmationMessage: 'ENG: Are you sure?',
                confirmButtonLabel: 'ENG: Yes',
            },
            showAlternateActionButton: true,
        });
        expect(getByText('ENG: Confirmation')).toBeInTheDocument();
        expect(getByText('ENG: Are you sure?')).toBeInTheDocument();
        expect(getByRole('button', { name: /ENG: Yes/ })).toBeInTheDocument();
        expect(getByRole('button', { name: /ENG: Maybe/ })).toBeInTheDocument();
        expect(getByRole('button', { name: /ENG: No/ })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it('the ok-equivalent button should work', () => {
        const testFn = jest.fn();
        const { getByRole } = setup({
            onAction: testFn,
        });
        fireEvent.click(getByRole('button', { name: /Yes/ }));
        expect(testFn).toHaveBeenCalled();
    });

    it('the cancel-equivalent button should work', () => {
        const testFn = jest.fn();
        const { getByRole } = setup({
            onCancelAction: testFn,
        });
        fireEvent.click(getByRole('button', { name: /No/ }));
        expect(testFn).toHaveBeenCalled();
    });

    it('the alternate action button should work', () => {
        const testFn = jest.fn();
        const { getByRole } = setup({
            showAlternateActionButton: true,
            onAlternateAction: testFn,
        });
        fireEvent.click(getByRole('button', { name: /Maybe/ }));
        expect(testFn).toHaveBeenCalled();
    });

    describe('onRef behavior', () => {
        it('calls onRef with instance on mount and undefined on unmount', () => {
            const onRef = jest.fn();
            const { unmount } = setup({ onRef });

            // on mount
            expect(onRef).toHaveBeenCalledWith(expect.any(ConfirmDialogBoxClass));
            // sanity check
            expect(onRef.mock.calls[0][0]).toHaveProperty('showConfirmation');
            // on unmount
            unmount();
            expect(onRef).toHaveBeenLastCalledWith(undefined);
        });
    });

    describe('Class instance', () => {
        const getInstance = props => {
            const instance = new ConfirmDialogBoxClass(props ?? defaultProps);
            instance.setState = jest.fn(newState => {
                instance.state = { ...instance.state, ...newState };
            });
            return instance;
        };

        it('should show and hide confirmation dialog', () => {
            const instance = getInstance({ ...defaultProps, isOpen: false });
            instance.showConfirmation();
            expect(instance.state.isDialogOpen).toBeTruthy();

            instance._hideConfirmation();
            expect(instance.state.isDialogOpen).toBeFalsy();
        });
    });
});
