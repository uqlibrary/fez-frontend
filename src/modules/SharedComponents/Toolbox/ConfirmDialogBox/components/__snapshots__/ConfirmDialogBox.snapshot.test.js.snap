// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogBox snapshots tests renders component with customised locale 1`] = `
<ForwardRef(Dialog)
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    ENG: Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      ENG: Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          ENG: Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          data-analyticsid="alternate-dialog-box"
          data-testid="alternate-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          ENG: Maybe
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          color="primary"
          data-analyticsid="cancel-dialog-box"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          ENG: No
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes 1`] = `
<ForwardRef(Dialog)
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes/no buttons 1`] = `
<ForwardRef(Dialog)
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          color="primary"
          data-analyticsid="cancel-dialog-box"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes/no/maybe buttons 1`] = `
<ForwardRef(Dialog)
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          data-analyticsid="alternate-dialog-box"
          data-testid="alternate-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Maybe
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          color="primary"
          data-analyticsid="cancel-dialog-box"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 1`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": "test3",
    },
    "backgroundColor": "test2",
    "color": "test4",
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 2`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": undefined,
    },
    "backgroundColor": undefined,
    "color": undefined,
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 3`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": undefined,
    },
    "backgroundColor": undefined,
    "color": undefined,
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should show and hide confirmation dialog 1`] = `
<ForwardRef(Dialog)
  open={true}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          color="primary"
          data-analyticsid="cancel-dialog-box"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;

exports[`ConfirmDialogBox snapshots tests should show and hide confirmation dialog 2`] = `
<ForwardRef(Dialog)
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <ForwardRef(DialogTitle)
    data-testid="message-title"
  >
    Confirmation
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)
      data-testid="message-content"
    >
      Are you sure?
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Grid)
      container={true}
      spacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
          }
        }
        xs={true}
      />
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          autoFocus={true}
          color="primary"
          data-analyticsid="confirm-dialog-box"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        sm="auto"
        xs={12}
      >
        <ForwardRef(Button)
          color="primary"
          data-analyticsid="cancel-dialog-box"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
